enhancements/20251015-support-ipconfig-on-clusternetworks.md

```diff
@@ -0,0 +1,413 @@
+- vlan sub interface `cluster-1-br.2012` is created on node 1,node 2 and node 3 in the cluster and ip address is assigned from DHCP.
+
+#### Story 2: Create VM Network with cluster network `cluster-2`, vid `2013`,`L3 Connectivity` field set and mode `DHCP`
+
+- bridge vlan vid `2013` added to `cluster-1-br` on node 1,node 2
+- vlan sub interface `cluster-1-br.2013` is created on node 1,node 2 in the cluster and ip address is assigned from DHCP.
+
+#### Story 3: Update VM Network with cluster network `cluster-1`, vid `2012`,`L3 Connectivity` field set and mode `Static`
+     "node 1": "192.168.1.10/24",
+     "node 2": "192.168.1.11/24",
+     "node 3": "192.168.1.12/24"
+
+A vlan sub interface `cluster-1-br.2012` is created on all 3 nodes in the cluster and ip address "192.168.1.10/24" assigned to interface `cluster-1-br.2012` of node 1,
+"192.168.1.11/24" assigned to `cluster-1-br.2012` of node 2 and "192.168.1.12/24" assigned to interface `cluster-1-br.2012` of node 3.
+
+#### Story 4: Update VM Network with cluster network `cluster-1`, vid `2012`,`L3 Connectivity` field unset
+
+- bridge vlan vid `2012` removed from `cluster-1-br` on node 1,node 2 and node 3
+- vlan sub interface `cluster-1-br.2013` is removed from node 1,node 2 and node 3
+
+#### Story 5: Delete VM Network with cluster network `cluster-1`, vid `2012`
+
+- bridge vlan vid `2012` removed from `cluster-1-br` on node 1,node 2 and node 3
+- vlan sub interface `cluster-1-br.2012` is removed from node 1,node 2 and node 3
+
+#### Story 6: Delete network config under cluster network `cluster-1`
+
+- Remove all vlan associated with `cluster-1` on node 1, node 2 and node 3
+- Remove all vlan sub interfaces associated with `cluster-1 from node 1,node 2 and node 3
+
+#### Story 7: New node node 4 added to the cluster
+
+- bridge vlan vid `2012` added to `cluster-1-br` on node 4
+- vlan sub interface `cluster-1-br.2012` is created on node 4 in the cluster and ip address is assigned from DHCP.
+
+#### Story 8: `Node 3` removed from network config of `cluster-1` or `Node 3` shutdown
+
+- bridge vlan vid `2012` removed from `cluster-1-br` on node 3
+- vlan sub interface `cluster-1-br.2012` is removed on node 3
+
+#### Story 9: New node node 4 added to the cluster (L3 connectivity mode `static`)
+
+- User reconfigures VM Network `2012` with node IP "192.168.1.13/24" for node 4
+- bridge vlan vid `2012` added to `cluster-1-br` on node 4
+- vlan sub interface `cluster-1-br.2012` is created on node 4 in the cluster and ip address "192.168.1.12/24" is assigned to `cluster-1-br.2012`
+
+#### Story 10: Reboot node 1, node 2 and node 3
+
+After the nodes come up,
+- bridge vlan vid `2012` added to `cluster-1-br` on node 1,node 2 and node 3
+- vlan sub interface `cluster-1-br.2012` is created on node 1,node 2 and node 3 in the cluster and ip address is assigned from DHCP or static
+
+#### Story 11: Update node selector in network config under cluster network `cluster-1` to node-1
```

Comment:
it depends on the upper layer, if OVN is sitting on this network, the change might break VMs